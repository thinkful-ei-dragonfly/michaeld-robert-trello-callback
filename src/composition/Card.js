import React from 'react';

function Card(props) {
console.log(props.id, 'This is card')

  return (
    <div className='Card'>
      <button onClick={() => props.deleteCard(props.id)}
        type='delete'>delete</button>
      <h3>{props.title}</h3>
      <p>{props.content}</p>
      
    </div>
  )
}

export default Card;