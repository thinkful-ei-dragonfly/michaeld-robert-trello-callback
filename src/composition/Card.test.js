import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card';
import '../index.css';
import STORE from '../store';

describe('<Card />', () => {
  it('renders without crashing', () => {
      // Create a DOM element to render the component into
      const div = document.createElement('div');
      // Render the component
      // If something is wrong it will fail here
      ReactDOM.render(<Card key={STORE.allCards.key} title={STORE.allCards.title} content={STORE.allCards.content}/>, div);
      // Clean up
    ReactDOM.unmountComponentAtNode(div);
  })
  it('renders without crashing', () => {
    // Render the component, as JSON
    const tree = renderer.create(<Card key={STORE.allCards.key} title={STORE.allCards.title} content={STORE.allCards.content}/>).toJSON();
    // Check whether it matches the previous snapshot
    // Stored in __snapshots__/App.test.js.snap
    expect(tree).toMatchSnapshot();
  });
});