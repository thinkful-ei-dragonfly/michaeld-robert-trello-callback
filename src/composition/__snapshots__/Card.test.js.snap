// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders without crashing 1`] = `
<div
  className="Card"
>
  <button
    type="delete"
  >
    delete
  </button>
  <h3 />
  <p />
</div>
`;
